src/cpp/src/visual_language/inputs_embedder.cpp

```diff
@@ -112,19 +128,30 @@ class InputsEmbedder::IInputsEmbedder {
                 new_templated_chat_history = m_tokenizer.apply_chat_template(m_history, add_generation_prompt, chat_template_fallback);
             }
             ov::Tensor new_chat_tokens = m_tokenizer.encode(new_templated_chat_history).input_ids;
-            if (m_is_cache_empty) {
+            TokenizedInputs prev_chat_tokens = m_tokenizer.encode(m_templated_chat_history);
+
+            // some symbols combinations can be encoded by the tokenizer in different ways
+            // if we met sequence with such combination of symbols, we cannot correctly subtract the new history from the old history
+            // and find the difference as a prompt
+            // so let's check it out and use the whole history in this case
+            if (!m_is_cache_empty && m_trust_encoded_history) {
+                m_trust_encoded_history = ov::genai::utils::is_tokenized_history_same(prev_chat_tokens.input_ids, m_tokenized_chat_history);
+            }
+
+            if (m_is_cache_empty || m_trust_encoded_history) {
```

Comment:
```suggestion
            if (m_is_cache_empty || !m_trust_encoded_history) {
```
because we need to take whole history if we cannot trust encoded history (KV cache) ?

Reply:
1. is it not addressed in current PR, right? just for my understanding 
2. whole history misses actual image tokens (their imbeddings) and I suppose it's a reason of incorrect answer.

Reply:
1. now no, with trimming KV cache it will be part from first different token to the end of the history . Regarding token 1576 I'll check, this is not first case when switching between symbols happens if something appears before the sequence